File: prosessering/src/main/java/no/nav/familie/prosessering/TaskBeskrivelse.java
package no.nav.familie.prosessering;


import java.lang.annotation.*;

@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Inherited
public @interface TaskBeskrivelse {

    /**
     * Antall retries på tasken før den settes til feilet
     *
     * @return antall forsøk
     */
    int maxAntallFeil() default 3;

    /**
     * Task typen
     *
     * @return typen
     */
    String taskType();

    /**
     * Beskrivelse for task typen
     *
     * @return beskrivelen
     */
    String beskrivelse();
}

File: prosessering/src/main/java/no/nav/familie/prosessering/TaskBeskrivelse.kt
package no.nav.familie.prosessering


import java.lang.annotation.*

@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Inherited
annotation class TaskBeskrivelse(
        /**
         * Antall retries på tasken før den settes til feilet
         *
         * @return antall forsøk
         */
        val maxAntallFeil: Int = 3,
        /**
         * Task typen
         *
         * @return typen
         */
        val tasktype: String,
        /**
         * Beskrivelse for task typen
         *
         * @return beskrivelsen
         */
        val beskrivelse: String)
